salt/metalk8s/orchestrate/upgrade/init.sls

```diff
@@ -10,6 +11,15 @@ Execute the upgrade prechecks:
         orchestrate:
           dest_version: {{ dest_version }}
 
+{% for minion in minions %}
+Refresh the package repository for {{ minion }}:
+  salt.function:
+    - tgt: {{ minion }}
+    - name: pkg.refresh_db
+    - watch:
+      - Execute the upgrade prechecks
+{% endfor %}
```

Comment:
No sorry I think you should do it in the `for` loop already here just below.

And again I'm not sure `pkg.refresh_db` is enough did you test it ?

Reply:
> No sorry I think you should do it in the `for` loop already here just below.
> 
OK, well noted

> And again I'm not sure `pkg.refresh_db` is enough did you test it ?
>
Currently bootstrapping a test env, I will let you know if this fixes the issue.

Reply:
Seems to work as we expected